elide-core/src/main/java/com/yahoo/elide/security/executors/ActivePermissionExecutor.java

```diff
@@ -90,64 +97,79 @@ public ActivePermissionExecutor(final com.yahoo.elide.core.RequestScope requestS
      * @see com.yahoo.elide.annotation.DeletePermission
      */
     @Override
-    public <A extends Annotation> void checkPermission(Class<A> annotationClass,
+    public <A extends Annotation> ExpressionResult checkPermission(Class<A> annotationClass,
                                                        PersistentResource resource,
                                                        ChangeSpec changeSpec) {
         if (requestScope.getSecurityMode() == SecurityMode.SECURITY_INACTIVE) {
-            return; // Bypass
+            return ExpressionResult.PASS; // Bypass
         }
         Expressions expressions;
         if (SharePermission.class == annotationClass) {
             expressions = expressionBuilder.buildSharePermissionExpressions(resource);
         } else {
             expressions = expressionBuilder.buildAnyFieldExpressions(resource, annotationClass, changeSpec);
         }
-        executeExpressions(expressions, annotationClass);
+        return executeExpressions(expressions, annotationClass);
     }
 
     /**
      * Check for permissions on a specific field.
      *
+     * @param <A> type parameter
      * @param resource resource
      * @param changeSpec changepsec
      * @param annotationClass annotation class
      * @param field field to check
-     * @param <A> type parameter
      */
     @Override
-    public <A extends Annotation> void checkSpecificFieldPermissions(PersistentResource<?> resource,
-                                                                     ChangeSpec changeSpec,
-                                                                     Class<A> annotationClass,
-                                                                     String field) {
+    public <A extends Annotation> ExpressionResult checkSpecificFieldPermissions(PersistentResource<?> resource,
+                                                                                 ChangeSpec changeSpec,
+                                                                                 Class<A> annotationClass,
+                                                                                 String field) {
         if (requestScope.getSecurityMode() == SecurityMode.SECURITY_INACTIVE) {
-            return; // Bypass
+            return ExpressionResult.PASS; // Bypass
         }
+
+        ExpressionResult expressionResult = this.checkUserPermissions(resource, annotationClass, field);
+        if (requestScope.getExpressionResultShortCircuit()
```

Comment:
It would be more natural to check for PASS here directly (by comparing with the expressionResult)